macros/regex/regexp_instr.sql

```diff
@@ -50,9 +50,31 @@ regexp_instr({{ source_value }}, '{{ regexp }}', {{ position }}, {{ occurrence }
 {% macro _validate_flags(flags, alphabet) %}
 {% for flag in flags %}
     {% if flag not in alphabet %}
-    {{ exceptions.raise_compiler_error(
+    {# Using raise_compiler_error causes disabled tests with invalid flags to fail compilation #}
+    {{ exceptions.warn(
         "flag " ~ flag ~ " not in list of allowed flags for this adapter: " ~ alphabet | join(", ")
     ) }}
     {% endif %}
 {% endfor %}
+{% endmacro %}
+
+{# Re2 requires specific flag validation because of its clear flag operator #}
+{% macro _validate_re2_flags(flags) %}
+{# Re2 supports following flags: #}
+{# i  :  case-insensitive (default fault) #}
+{# m  :  multi-line mode: ^ and $ match begin/end line in addition to begin/end text (default false) #}
+{# s  :  let . match \n (default false) #}
+{# U  :  ungreedy: swap meaning of x* and x*?, x+ and x+?, etc (default false) #}
+{# Flag syntax is xyz (set) or -xyz (clear) or xy-z (set xy, clear z).  #}
+
+{# Regex explanation: do not allow consecutive dashes, accept all re2 flags and clear operator, do not end with a dash  #}
+{% set re2_flags_pattern = '^(?!.*--)[-imsU]*(?<!-)$' %}
+{% set re = modules.re %}
+{% set is_match = re.match(re2_flags_pattern, flags) %}
+{% if not is_match %}
+    {# Using raise_compiler_error causes disabled tests with invalid flags to fail compilation #}
+    {{ exceptions.warn(
+        "flags " ~ flags ~ " isn't a valid re2 flag pattern" 
```

Comment:
if a user passes in an unsupported flag, like `c` on BigQuery, this will generate a compiler warning, but the test will ultimately fail (e.g. "Cannot parse regular expression: invalid perl operator: (?c"). Is that the intent?

Reply:
The issue if we raise a compiler error is that the tests don't compile anymore.
As you can [see in the pipeline test](https://app.circleci.com/pipelines/github/calogica/dbt-expectations/237/workflows/e8184c91-8b40-4e48-8cc7-fef33e34c0b0/jobs/156?invite=true#step-106-9), even though the test for the "c" flag is not enabled for BigQuery (since it isn't a valid re2 flag), we have a warning in the output.
```
02:04:16  flags c isn't a valid re2 flag pattern
02:04:16  flags c isn't a valid re2 flag pattern
02:04:16  flags c isn't a valid re2 flag pattern
```
Here's the output with raise_compiler_error instead of warn:
```
Encountered an error:
Compilation Error in test dbt_expectations_expect_column_values_to_match_regex_data_text_email_address__c___A_Z_ (models/schema_tests/schema.yml)
  flags c isn't a valid re2 flag pattern
```
Having a warning is nice since the test will compile properly but it can be forced to error with the `--warn-error` flag.
i.e.:
running  `dbt --warn-error build` will fail build for invalid flags in test (even if the invalid-flag tests are disabled).
running  `dbt build` will build for invalid flags in test (but display a warning message and ultimately fail)

However, I get that this might be considered a regression since it changes the default behavior to be more lax. 
This gives us 2 options:
1. Remove the "c" flag test
2. Add support for the "c" flag in BigQuery by translating it to "-i". This improves adapter flexibility but might be unintuitive for the user since "c" is not supposed be a valid re2 flag.

Reply:
Sorry for the delayed response. I think it's to leave as is, including the c flag tests with the defined targets. If we find that this creates usability issues for folks, we can address in a later PR (and maybe implement option 2). Thanks for the thorough explanation!